pkg/util/jsonpatch/jsonpatch.go

```diff
@@ -0,0 +1,135 @@
+package jsonpatch
+
+import (
+	"encoding/json"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// PatchBuilder is an interface for building a set of changes to a JSON document.
+type PatchBuilder interface {
```

Comment:
Why do you need it to be an interface if there is only one implementation of it (`type patch struct {`). Are you thinking on some potential structs in the future which could use it?

Reply:
Yeah, at one point I had the WithPath a d LabelSet stuff in separate types but then I merged them into the base.

Kubernetes also has a lot of interfaces with only one implementation, I might be picking up bad habits from reading their code.